cryptolib/signer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "errors"
+
+// Signer contains methods to create a signed Attestation.
+type Signer interface {
+	CreateAttestation(privateKey, payload []byte, publicKeyID string, alg SigningAlgorithm, keyMode KeyMode) (*Attestation, error)
```

Comment:
Document all function in the interface. Individual implementations document things specific to the implementation and can reference the documentation in the interface. Ditto all interfaces in this PR

Reply:
Done.